src/error.rs

```diff
@@ -27,6 +27,10 @@ pub enum Error {
     InvalidCredentials,
     /// Validation error when constructing the request or parsing the response.
     InvalidHttpFormat(http::Error),
+    /// Invalid HeaderName
+    InvalidHttpHeaderName(http::header::InvalidHeaderName),
+    /// Invalid HeaderValue
+    InvalidHttpHeaderValue(http::header::InvalidHeaderValue),
```

Comment:
Adding new variants to this enum unfortunately a breaking change, which I'd prefer to avoid if possible. Since `http::Error` implements `From<http::header::InvalidHeaderName>` and `From<http::header::InvalidHeaderValue>`, I recommend producing an `Error::InvalidHttpFormat` for now.